src/index.js

```diff
@@ -0,0 +1,266 @@
+    if (temp === 49 && state.realTempValue<= temp){
+      colorData.realTempValueColor = tempProperties[idx].color;
+      break;
+    }else if (state.realTempValue >= temp){
+      colorData.realTempValueColor = tempProperties[idx].color;
+      break;
+    }
+  }
+  document.getElementById('realTempValue').classList.remove(state.realTempValueColor)
+  updateState(colorData);
+  document.getElementById('realTempValue').classList.add(state.realTempValueColor)
+
+};
+
+const changeRealWeatherDetails = () =>{
+  const weatherDetails = document.getElementById('weather-details');
+  weatherDetails.textContent = state.weatherDescription[0].toUpperCase() + state.weatherDescription.substring(1);
+
+  const weatherIcon = document.getElementById('weather-icon');
+  if (state.weatherIconCode != '??'){
+    weatherIcon.src = `https://openweathermap.org/img/wn/${state.weatherIconCode}@2x.png`;
+  } else {
+    weatherIcon.src = 'assets/unknown-weather-icon.png'
+  }
+  };
+
+const increaseTemp = () =>{
+  state.gardenTempValue += 1;
+  const temp = document.getElementById('tempValue');
+  temp.textContent = state.gardenTempValue;
+  changeGardenTempValueColorAndLandscape();
+};
+
+const decreaseTemp = () =>{
+  state.gardenTempValue -= 1;
+  const temp = document.getElementById('tempValue');
+  temp.textContent  = state.gardenTempValue;
+  changeGardenTempValueColorAndLandscape();
+};
+
+
+// Wave 3
+const updateCityName = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  state.name = cityNameInput.value;
+  
+  const headerCityName = document.getElementById("headerCityName");
+  if (state.name != ''){
+    headerCityName.textContent = state.name[0].toUpperCase() + state.name.substring(1);
+  };
+};
+
+// Wave 4
+const getCityCoords = () =>{
+  const city = state.name;
+  return axios
+    .get('https://ada-weather-report-proxy-server.onrender.com/location', {params:{q: city}})
+    .then((response)=>{
+      const results = {
+        cityLat:response.data[0].lat,
+        cityLon: response.data[0].lon
+      };
+      return results;
+    })
+    .catch((error) => HandleCityNameError());
+}
+
+const getCityWeatherData = (coordObject) =>{
+  return axios
+    .get ('https://ada-weather-report-proxy-server.onrender.com/weather', {params:{lat: coordObject.cityLat, lon: coordObject.cityLon}})
+    .then((response)=>{
+      const tempK = response.data.main.temp;
+      const tempF = (tempK - 273.15) * 1.8 + 32;
+      const weatherData ={
+        realTempValue: parseInt(tempF.toFixed(0)),
+        weather: response.data.weather[0].main,
+        weatherIconCode:  response.data.weather[0].icon,
+        weatherDescription: response.data.weather[0].description,
+      };
+      return weatherData;
+    })
+    .catch((error) => HandleCityNameError());
+  };
+  
+  
+  const getCityData =  () => {
+    return getCityCoords()
+    .then( (coordList) => {
+      return getCityWeatherData(coordList);
+    })
+  };
```

Comment:
Really nice use of helper functions to divide up the actions we want to take into smaller portions that are easier to read and test.